design/deployment_design.md

```diff
@@ -0,0 +1,70 @@
+#  **Dockerless CLI -** deployment
+
+This document is the design document for the deployment of the Python CLI. 
+
+Currently, our Python CLI can be deployed in the following ways:
+
+1. Install the CLI via the pip Python package manager like so: `pip3 install conjur-client`. If installed in this way, Python **is** required on machine. If installed via pip, the user can then run the CLI like so `conjur-cli --help`
+
+2. Install and run the executable (Python **is** **not** required on the machine)
+
+   The CLI is delivered as a standalone executable, packed with the required runtime and dependencies. Therefore, there is no need for Python or other prerequisites to be installed on machine. The executable can be fetched by navigating to our [Python CLI release page](https://github.com/cyberark/conjur-api-python3/releases/tag/v0.0.5). If installed in this manner, `./python-cli-executable-location variable get ...`
+
+### Design
+
+#### Pip3 package
+
+We make CLI package available to the public by pushing to a public repository called PyPi. I propose that we maintain pushing our the Python CLI to pip3 for both customer's machines have have internet access and for our development work. The infrastructure is already in-place so this would be at no additional effort.
```

Comment:
```suggestion
We make CLI package available to the public by pushing to a public repository called PyPi. I propose that we maintain pushing our the Python CLI to pip3 for both customer's machines have have internet access and for our development work. Our Python API is also delivered this way and PyPi is the main installation method for it. The infrastructure is already in-place so this would be at no additional effort.
```

Reply:
@sgnn7 What are the implication for having the Python API?
Can we not deliver it?

Reply:
I was just noting here that the python client library is delivered to our users via PyPi too and it's the [primary way of using it](https://github.com/cyberark/conjur-api-python3#from-pypi). We must retain PyPi publishing if we want that to continue.